simple/rdm_tagged_peek.c

```diff
@@ -31,95 +31,187 @@
 #include <stdlib.h>
 #include <string.h>
 #include <getopt.h>
+#include <unistd.h>
 
 #include <rdma/fi_errno.h>
 #include <rdma/fi_tagged.h>
 
 #include <shared.h>
 
-static int tagged_peek(uint64_t tag)
+static int wait_for_send_comp(int count)
 {
+	int ret, completions = 0;
 	struct fi_cq_tagged_entry comp;
-	struct fi_msg_tagged msg;
+
+	do {
+		ret = fi_cq_sread(txcq, &comp, 1, NULL, -1);
```

Comment:
This actually exposed a bug in the sockets provider.  The provider should really handle the EAGAIN case internally, at least in most cases, and continue to wait.  I'd prefer to keep the code as is.